src/visitors/rules/custom/parameterAllOf.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable class-methods-use-this */
+
+class parameterAllOf {
+  constructor(config) {
+    this.config = { ...config };
+    switch (this.config.level) {
+      case 'warning':
+        this.config.level = 3;
+        break;
+      case 'error':
+      default:
+        this.config.level = 4;
+        break;
+    }
+  }
+
+  get rule() {
+    return 'parameterAllOf';
+  }
+
+  OpenAPIParameter() {
+    return {
+      onEnter: (node, _, ctx, unresolved, traverseTools) => {
```

Comment:
how to add an error from this validator? Could you add an example of validating the description field to be string (using custom code)

Reply:
for now we allow types (re)definition, not so much custom validators. we can (not sure if already) load additional rules (with are the same syntax/structure) as our own ones and run them on types we want